core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+          expect(
+              CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([]);
+        });
+
+    it('should correctly remove the first node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id0');
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, thirdCollectionNode]);
+    });
+
+    it('should correctly remove the last node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id2');
+      expect(collection.containsCollectionNode('exp_id2')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+    });
+
+    it('should correctly remove a middle node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id1');
+      expect(collection.containsCollectionNode('exp_id1')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode]);
+    });
+  });
+
+  describe('Collection linearizer node append function', function() {
+    it('should correctly append a node in an empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id0',
+          firstCollectionNode.getExplorationSummaryObject());
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([firstCollectionNode]);
+    });
+
+    it('should correctly append a node in a non-empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      collection.addCollectionNode(firstCollectionNode);
+      collection.addCollectionNode(secondCollectionNode);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id2',
+          thirdCollectionNode.getExplorationSummaryObject());
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+    });
+  });
+
+  describe('Collection linearizer node shifting function', function() {
```

Comment:
Some additional test ideas:

- Shifting a non-existent exploration left
- Shifting a non-existent exploration right
- Shifting a node left in an exploration with a single node
- Shifting a node right in an exploration with a single node
- Shifting a node left when there is a node to the left of it, but not to the right of it
- Shifting a node left when there is a node to the right of it, but not to the left of it
- Shifting a node right when there is a node to the left of it, but not to the right of it
- Shifting a node right when there is a node to the right of it, but not to the left of it

The above are testing extreme cases and also ensuring the three cases outlined in the implementation code are covered by separate unit tests:

```javascript
      // There are three cases when swapping a node left:
      //   1. There is a node right of it but not left
      //   2. There is a node left of it but not right
      //   3. There are nodes both to the left and right of it
```